lib/blob.js

```diff
@@ -1044,17 +1193,957 @@ Blob.prototype.leaseContainer = function leaseContainer(name, options) {
+ *    ifMatch: '...',                                         // ETag value. Specify this to perform the operation only if the resource's ETag matches the value specified. (optional)
+ *    ifNoneMatch: '...',                                     // ETag value. Specify this to perform the operation only if the resource's ETag does not match the value specified. (optional)
+ * }
+ *```
+ *
+ * @return {Promise} A promise for an object on the form:
+ * ```js
+ * {
+ *      eTag: '...',               // The entity tag of the blob
+ *      lastModified: '...',       // The date/time the blob was last modified
+ *      blobSequenceNumber: '...', // The blob's current sequence number (if the blob is a page blob)
+ * }
+ * ```
+ */
+Blob.prototype.setBlobProperties = function setBlobProperties(container, blob, options) {
+  assert(typeof container === 'string', 'The name of the container must be specified and must be a string value.');
+  assert(typeof blob === 'string', 'The name of the blob must be specified and must be a string value.');
+
+  var query = {
+    comp: 'properties'
+  };
+  var path = '/' + container + '/' + blob;
+  var headers = {};
+
+  if (options){
+    if (options.cacheControl) headers['x-ms-blob-cache-control'] = options.cacheControl;
+    if (options.contentType) headers['x-ms-blob-content-type'] = options.contentType;
+    if (options.contentMD5) headers['x-ms-blob-content-md5'] = options.contentMD5;
+    if (options.contentEncoding) headers['x-ms-blob-content-encoding'] = options.contentEncoding;
+    if (options.contentLanguage) headers['x-ms-blob-content-language'] = options.contentLanguage;
+    if (options.contentDisposition) headers['x-ms-blob-content-disposition'] = options.contentDisposition;
+    if (options.pageBlobContentLength) headers['x-ms-blob-content-length'] = options.pageBlobContentLength;
+    if (options.pageBlobSequenceNumberAction){
+      assert(options.pageBlobSequenceNumberAction === 'max'
+        || options.pageBlobSequenceNumberAction === 'update'
+        || options.pageBlobSequenceNumberAction === 'increment',
+        'The `options.pageBlobSequenceNumberAction` is invalid. The possible values are: max, update and increment.');
+      headers['x-ms-sequence-number-action'] = options.pageBlobSequenceNumberAction;
+      if (options.pageBlobSequenceNumberAction === 'max'
+        || options.pageBlobSequenceNumberAction === 'update'
+        && !options.pageBlobSequenceNumber) {
+        throw new Error('If `options.pageBlobSequenceNumberAction` is `max` or `update`, the `options.pageBlobSequenceNumber` must be supplied.');
+      }
+      if (options.pageBlobSequenceNumber) headers['x-ms-blob-sequence-number'] = options.pageBlobSequenceNumber;
+    }
+
+    utils.setConditionalHeaders(headers, {
+      ifModifiedSince: options.ifModifiedSince,
+      ifUnmodifiedSince: options.ifUnmodifiedSince,
+      ifMatch: options.ifMatch,
+      ifNoneMatch: options.ifNoneMatch
+    });
```

Comment:
I agree with you! There are 4 conditional headers that can be set, but not all the methods have support for all of them (e.g. for setContainerMetadata, you can set only the 'if-modified-since' header). Because of this, I decided to send a new object with properties representing only the headers that are supported by that method. I wanted to avoid the possibility of setting wrong headers.